server/src/main/java/org/elasticsearch/search/rank/context/RankFeaturePhaseRankCoordinatorContext.java

```diff
@@ -72,13 +81,18 @@ public void computeRankScoresForGlobalResults(
     }
 
     /**
-     * Ranks the provided {@link RankFeatureDoc} array and paginates the results based on the `from` and `size` parameters.
+     * Ranks the provided {@link RankFeatureDoc} array and paginates the results based on the `from` and `size` parameters. Filters out
+     * documents that have a relevance score less than min_score.
+     * @param rankFeatureDocs documents to process
      */
     public RankFeatureDoc[] rankAndPaginate(RankFeatureDoc[] rankFeatureDocs) {
-        Arrays.sort(rankFeatureDocs, Comparator.comparing((RankFeatureDoc doc) -> doc.score).reversed());
-        RankFeatureDoc[] topResults = new RankFeatureDoc[Math.max(0, Math.min(size, rankFeatureDocs.length - from))];
+        RankFeatureDoc[] rankFeatureDocsAboveMinScore = Arrays.stream(rankFeatureDocs)
+            .filter(this::keepRankFeatureDoc)
```

Comment:
this is misnamed, its about if we keep the feature doc or not. There is nothing in particular about "min score" here. All the concern about min score seems to be in the sub class.

Reply:
[Refactored](https://github.com/elastic/elasticsearch/pull/109813/commits/62ab07cb299af3ecf551fa2021c2542c940b9271) - I prefer to move `min_score`-based logic into the subclass, since it's not guaranteed that every `RankFeaturePhaseRankCoordinatorContext` implementation will use scores.